### PR TITLE
Update to latest byron and shelley ledgers

### PR DESCRIPTION
**Pull request by dcoutts**
And latest cardano-base. The base and shelley ledger changes include some changes to the annotations.

The changes in the code in this patch are due to Shelley ledger changes.

There is one part I cannot figure out easily and is left as an exercise to the reader.

### CODE REVIEW

File: ouroboros-consensus-shelley/src/Ouroboros/Consensus/Shelley/Ledger/Mempool.hs [hunk 1]
```diff
@@ -95,6 +96,11 @@ instance Crypto c => ToCBOR (GenTx (ShelleyBlock c)) where
 instance Crypto c => FromCBOR (GenTx (ShelleyBlock c)) where
   fromCBOR = mkShelleyTx <$> fromCBOR
 
+-- TODO: the above instance needs to be updated but it's not immediately
+-- obvious to me how.
+-- It currently relies on an instance FromCBOR (SL.Tx crypto)
+-- but the ledger now provides the instance FromCBOR (Annotator (Tx crypto))
```
Comment (dcoutts) [inline ouroboros-consensus-shelley/src/Ouroboros/Consensus/Shelley/Ledger/Mempool.hs:102]: This bit.

**Comment by dcoutts** [comment]
This is needed for https://github.com/input-output-hk/cardano-node/pull/798

**Comment by mrBliss** [comment]
I had #1928 ready for this, but that depends on #1947. Can we merge those two PRs first?

Multiple repos + releasing from master gives this kind of problem. At some point we'll need release branches.

**Comment by dcoutts** [comment]
This is not needed for a release. Go with your original plan. We have a temporary workaround.

**Pull request closed without merging**